engine/engine.go

```diff
@@ -20,6 +20,7 @@ import (
 	"github.com/prometheus/prometheus/storage"
 	"github.com/prometheus/prometheus/util/annotations"
 	"github.com/prometheus/prometheus/util/stats"
+	v1 "github.com/prometheus/prometheus/web/api/v1"
```

Comment:
Hmm was this interface moved back?

Reply:
Related PR: https://github.com/prometheus/prometheus/pull/13713

Reply:
Should we perspectively move fallback into the consumers of this library? i.e. make thanos use prometheus engine if creation of a query fails with thanos engine? That way we can skip fallback and prometheus engine here.

Reply:
nvm, you were right. I was not taking `main`.